packages/server/src/app/handlers/handle-wallet-message.ts

```diff
@@ -0,0 +1,46 @@
+import { RelayableAction } from 'magmo-wallet/lib/src/communication';
+import { MessageRelayRequested } from '../../../../magmo-wallet-client/lib';
+import { getProcess } from '../../wallet/db/queries/walletProcess';
+import { handleNewProcessAction } from './handle-new-process-action';
+import { handleOngoingProcessAction } from './handle-ongoing-process-action';
+
+export async function handleWalletMessage(
+  message: RelayableAction,
+): Promise<MessageRelayRequested | undefined> {
+  if (await isNewProcessAction(message)) {
+    return handleNewProcessAction(message);
+  } else if (await isProtocolAction(message)) {
+    return handleOngoingProcessAction(message);
+  } else {
+    return undefined;
+  }
+}
+
+function isNewProcessAction(action: RelayableAction): boolean {
+  if (action.type === 'WALLET.NEW_PROCESS.CONCLUDE_INSTIGATED') {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+async function isProtocolAction(action: RelayableAction): Promise<boolean> {
```

Comment:
Again, perhaps this should be a type guard. (We can move the async code to the block at L13.